File: Daily Challenge/April 2025/Sum of All Subset XOR Totals.cs
public class Solution {
  public int SubsetXORSum(int[] nums) {
    int n = nums.Length;
    int totalXorSum = 0;
    
    for (int i = 0; i < (1 << n); i++) {
      int subsetXor = 0;
      for (int j = 0; j < n; j++) {
        if ((i & (1 << j)) != 0) {
          subsetXor ^= nums[j];
        }
      }
      totalXorSum += subsetXor;
    }

    return totalXorSum;
  }
}

File: Daily Challenge/April 2025/Sum of All Subset XOR Totals.cpp
class Solution {
public:
  int subsetXORSum(vector<int>& nums) {
    int n = nums.size();
    int totalXORSum = 0;

    for (int i = 0; i < (1 << n); ++i) {
      int currentXOR = 0;
      for (int j = 0; j < n; ++j) {
        if (i & (1 << j)) {
          currentXOR ^= nums[j];
        }
      }
      totalXORSum += currentXOR;
    }

    return totalXORSum;
  }
};